### ISSUE TITLE
[REQ] New Sticker System

### ISSUE THREAD

**Issue by crashzk** [ianlucas/cs2-inventory-simulator#17]
Can we expect a rewrite of the sticker system since in CS2 it was completely redone?

Furthermore, I noticed that when we apply stickers to weapons and even when we view weapons with stickers, they are not visible on the weapon.

![image](https://github.com/ianlucas/cs2-inventory-simulator/assets/32937653/20c2b580-afe1-4d0a-bc4a-10a9eb0fdc99)

**Comment by ianlucas** [comment]
Yes, I have some ideas for this system, but I need a little bit of time to work on it. Regarding previewing stickers, it's kinda #6, but something that can be implemented.

If someone releases a public CS# plugin for adding stickers that will be helpful to the implementation of the system, let's hope for that.

**Comment by crashzk** [comment]
Some features that can be added when the new sticker system is implemented.

Create rules to:
- Allow/Deny sticker rotation
- Allow/Deny sticker positioning 

https://twitter.com/i/status/1755016576313729398

**Comment by Liubik** [comment]
https://github.com/Nereziel/cs2-WeaponPaints/issues/238

skimmed through this, if this could be added (if it is possible) this would be crazy

**Comment by ianlucas** [comment]
> [Nereziel/cs2-WeaponPaints#238](https://github.com/Nereziel/cs2-WeaponPaints/issues/238)
> 
> skimmed through this, if this could be added (if it is possible) this would be crazy

Yeah I'm aware of this and I have an idea on how implement it, but currently I'm focusing on stabilizing what we have right now and refactoring some stuff. Hopefully by the end of April I can tackle this if I find more time and motivation.

**Comment by crashzk** [comment]
News about this feature?

**Comment by ianlucas** [comment]
No updates, still trying to find some motivation. I'm not sure the fifth sticker has a preset X and Y pos since it was added with the sticker positioning system, so adding it to the UI (and plugin) will probably cause issues (or not make sense without the positioning system in place).

**Comment by KewaiiGamer** [comment]
how is work on this? have you found the motivation? I find this project amazing and I am looking forward to more progress on it <3

**Comment by ianlucas** [comment]
Currently this is blocked as it isn't as simple as I thought it would be. Basically each weapon model has the offsets for each sticker position in its DATA section (see Source2Viewer). We need to figure out how that translates in the 3D space so that we can make the positioning exactly as in the game.

Eventually I may figure this out (we're talking years here), but for accelerating this process I'm happy to accept donations OR volunteers to research the matter (or open-source references).

**I KNOW MY WAY WITH 3D PROGRAMMING, HOW CAN I HELP?**

We'll be using Three.js for building the interface for placing the stickers, so if you're able to build a POC using the offsets of the Dual Berettas and demonstrate it's being translated correctly as in-game, we'll make great progress on this. I'm asking for Dual Berettas because it seems more trickier to figure out.

**Comment by crashzk** [comment]
Note: https://github.com/L1teD/cs2-WeaponPaints-website

From what I've seen some users mention, the private version uses stickers and charms, I can't confirm how the preview works, its demo is offline.

**Comment by KewaiiGamer** [comment]
I tried donating with paypal but it doesn't work. it gets stuck on button "Agree & Donate Now".

I really want to donate as this project as been the base for my cs2 community server website and it saved me several months of development.

**Comment by ianlucas** [comment]
> I tried donating with paypal but it doesn't work. it gets stuck on button "Agree & Donate Now".
> 
> I really want to donate as this project as been the base for my cs2 community server website and it saved me several months of development.
> 
> Also do you have a discord server or dm where we can talk?

I'm ianlucas in AM and CSSharp Discords, just send me more details of the issue you're facing with PayPal there and we can sort it out, thanks.

**Comment by sikkgit** [comment]
Hi! As a workaround, would it be possible to leverage inspect links from csinspect.com website and import them into the inventory?
For example, this Desert Eagle | Night with unique sticker placements has this console command: 

`csgo_econ_action_preview 00180120282802389AB3E6F00340016214080110AD232D06B186433D30F207BF457051CCBC6214080110AD232D35FB86433D67E2F6BE45C67E1D3D6214080110AD232DB62B87433D88F8B5BE45143D193D6214080110AD232D42FC86433DEC0AD0BE45AFB910BD6214080110AD232D992C87433DAD137ABE4598F4033DE1C84AFE`

![Image](https://github.com/user-attachments/assets/dfe163f3-52bd-4bf7-832b-fc4e9b78febf)

Granted, we wouldn't be able to edit this item or at least not the sticker positions but it would be better than nothing.

**Comment by ianlucas** [comment]
I'm making great progress on this, and I believe it can be ready to ship in about three weeks.

I'll also add an embeddable component that third-party websites can use.

It's wild to have it open source, considering the effort it took (hacky tho), but it's definitely better than gatekeeping it.

Please consider donating to the project. 🥲

**Comment by KewaiiGamer** [comment]
Will it have 3d rendering of model? 

also yes i'm considering donating but right now I'm in the process of buying a new house so as soon as it's solved I will definitely look into donating. i really love all your work in this project <3

**Comment by ianlucas** [comment]
Yes, it'll be 3D tho it's a bit hacky, but should look better than 2D canvas.

**Comment by ianlucas** [comment]
No news, I'm lacking motivation to focus on this right now. Donations are always welcome though.

**Comment by GL1TCH1337** [comment]
https://github.com/LielXD/CS2-WeaponPaints-Website got updated with 3d sticker preview system. I dont know which stage are you at but maybe this can help a bit.

**Comment by ianlucas** [comment]
Okay, I've got some updates on this feature, it should land in a month or so.
To start, it won't be open-source, instead it'll run as a hosted service at 3d.cstrike.app. Anyone will be able to use it in their own Inventory Simulator instance, though the output will carry a watermark. I've spent (and am still spending) a fair amount of cash on this, so public use may come with some rate limits... but I'll try to offer a way, via donations, for anyone who wants to use it without them.

https://github.com/user-attachments/assets/33bcf5e5-b2c7-4a04-9451-a8f893a6be05
